monkey/infection_monkey/monkey.py

```diff
@@ -375,7 +375,7 @@ def _build_victim_host_factory(
 
     def _running_on_island(self, local_network_interfaces: List[NetworkInterface]) -> bool:
         server_ip, _ = address_to_ip_port(self._control_client.server_address)
-        return server_ip in {interface.address for interface in local_network_interfaces}
+        return server_ip in {interface.ip.compressed for interface in local_network_interfaces}
```

Comment:
I would prefer str(interface.ip) to interface.ip.compressed or interface.ip.exploded. What does "compressed" mean? How is it different from "exploded"?